runtime-transaction/src/runtime_transaction/transaction_view.rs

```diff
@@ -91,6 +100,47 @@ impl<D: TransactionData> RuntimeTransaction<ResolvedTransactionView<D>> {
     }
 }
 
+impl<D: TransactionData> TransactionWithMeta for RuntimeTransaction<ResolvedTransactionView<D>> {
+    fn as_sanitized_transaction(&self) -> impl Borrow<SanitizedTransaction> {
+        let VersionedTransaction {
+            signatures,
+            message,
+        } = self.to_versioned_transaction();
+
+        let is_writable_account_cache = (0..self.transaction.total_num_accounts())
+            .map(|index| self.is_writable(usize::from(index)))
+            .collect();
+
+        let message = match message {
+            VersionedMessage::Legacy(message) => SanitizedMessage::Legacy(LegacyMessage {
+                message: Cow::Owned(message),
+                is_writable_account_cache,
+            }),
+            VersionedMessage::V0(message) => SanitizedMessage::V0(LoadedMessage {
+                message: Cow::Owned(message),
+                loaded_addresses: Cow::Owned(self.loaded_addresses().unwrap().clone()),
+                is_writable_account_cache,
+            }),
+        };
+
+        // SAFETY:
+        // - Simple conversion between different formats
+        // - `ResolvedTransactionView` has undergone sanitization checks
+        unsafe {
+            SanitizedTransaction::new_from_parts(
```

Comment:
The only alternative here is to pass the reserved_account_keys in and redo SanitizedTransacton checks